yb-voyager/cmd/getDataMigrationReportCommand.go

```diff
@@ -316,42 +332,54 @@ func updateImportedEventsCountsInTheRow(sourceDBType string, row *rowData, table
 	}
 	if importerRole == SOURCE_REPLICA_DB_IMPORTER_ROLE {
 		var err error
-		tblName := tableNameTup.ForKey()
-		tableNameTup, err = nameRegistryForSourceReplicaRole.LookupTableName(tblName)
-		if err != nil {
-			return fmt.Errorf("lookup %s in source replica name registry: %v", tblName, err)
+		sourceReplicaTups := make([]sqlname.NameTuple, len(tableNameTups))
+		for i, ntup := range tableNameTups {
+			sourceReplicaTups[i], err = namereg.NameReg.LookupTableName(ntup.ForKey())
+			if err != nil {
+				return nil, fmt.Errorf("lookup table %s: %v", ntup.ForKey(), err)
+			}
 		}
+		tableNameTups = sourceReplicaTups
 	}
-
 	//reinitialise targetDB
 	tconf = *targetConf
 	tdb = tgtdb.NewTargetDB(&tconf)
 	err := tdb.Init()
 	if err != nil {
-		return fmt.Errorf("failed to initialize the target DB: %w", err)
+		return nil, fmt.Errorf("failed to initialize the target DB: %w", err)
 	}
 	defer tdb.Finalize()
 	state := NewImportDataState(exportDir)
+	tableNameTupToEventsCounter, err := state.GetImportedEventsStatsForTableList(tableNameTups, migrationUUID)
+	if err != nil {
+		return nil, fmt.Errorf("imported events stats for tableList: %s", err)
+	}
+	return tableNameTupToEventsCounter, nil
+}
+
+func updateImportedEventsCountsInTheRow(row *rowData, tableNameTup sqlname.NameTuple, snapshotImportedRowsMap *utils.StructMap[sqlname.NameTuple, int64], eventsImportedMap *utils.StructMap[sqlname.NameTuple, *tgtdb.EventCounter]) error {
+	switch row.DBType {
+	case "target":
+		importerRole = TARGET_DB_IMPORTER_ROLE
+	case "source-replica":
+		importerRole = SOURCE_REPLICA_DB_IMPORTER_ROLE
+	case "source":
+		importerRole = SOURCE_DB_IMPORTER_ROLE
+	}
+	if importerRole == SOURCE_REPLICA_DB_IMPORTER_ROLE {
+		var err error
```

Comment:
Pls add a comment here as well. I forgot why exactly we need a separate name registry in this case...

Since we are looking up nametuple from the map, wouldn't it work even without changing the tableNameTup to be from the nameRegistryForSourceReplicaRole? i.e. because the map.Get would use ForKey, it should work even if the nametuples are different no?